doc/binaryrecord-spec.md

```diff
@@ -178,9 +186,9 @@ Note that map fields must be presorted before being added.
 
 Note that this is called a "Map" field but is actually just a list of key-value pairs.  Since none of the operations above involve actual lookup by key, O(1) lookup is not needed, plus the usually small number of keys means it is extremely fast to iterate through everything.
 
-* +0000   4 bytes  total length of map field not including these bytes
-* +0004   2-byte Length of key #1, or 0xFzzz for preset key field where zzz = preset number (up to 4K presets)
-* +0006 to +0006+(keylen1 - 1)   UTF8 bytes for key #1
+* +0000   2 bytes  total length of map field not including these bytes
+* +0002   1-byte Length of key #1, or if MSB is set, then preset key field where zzz = preset number (up to 128 presets)
+* +0003 to +0003+(keylen1 - 1)   UTF8 bytes for key #1, or no bytes if preset key
```

Comment:
Instead of packing everything tightly, would it make sense to leave some room for backward compatible additions in the future? As it stands, any change here would break protocol.

Reply:
The thoughts are the following:
1) If we really need more space for a map, then we can create a different `LargeMapColumn`, that is how compatibility would work.
2) It is true this is a format change, but the schemaID addition already is as well a breaking change.  No way around it.   In the future we can add column types if desired to change field format significantly.  

The observation is because virtually all of the maps we have today fits into the requirement, so why waste space?